### ISSUE TITLE
Unknown file extension ".json" 

### ISSUE THREAD

**Issue by damascene** [jacany/nblog#46]
I'm having the following error when I try to run `yarn build`:

```
TypeError [ERR_UNKNOWN_FILE_EXTENSION]: Unknown file extension ".json" for /home/user/nostr_stuff/nblog/package.json
```

I've run the instructions as in in the README.md page.

```
$ yarn
yarn install v1.22.19
warning ../../../package.json: No license field
[1/4] Resolving packages...
[2/4] Fetching packages...
[3/4] Linking dependencies...
[4/4] Building fresh packages...
Done in 29.06s.

$ yarn build
yarn run v1.22.19
warning ../../../package.json: No license field
$ vite build
▲ [WARNING] Cannot find base config file "./.svelte-kit/tsconfig.json" [tsconfig.json]

    tsconfig.json:2:12:
      2 │   "extends": "./.svelte-kit/tsconfig.json",
        ╵              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

error during build:
TypeError [ERR_UNKNOWN_FILE_EXTENSION]: Unknown file extension ".json" for /home/user/nostr_stuff/nblog/package.json
    at new NodeError (node:internal/errors:371:5)
    at Object.getFileProtocolModuleFormat [as file:] (node:internal/modules/esm/get_format:87:11)
    at defaultGetFormat (node:internal/modules/esm/get_format:102:38)
    at defaultLoad (node:internal/modules/esm/load:21:14)
    at ESMLoader.load (node:internal/modules/esm/loader:359:26)
    at ESMLoader.moduleProvider (node:internal/modules/esm/loader:280:58)
    at new ModuleJob (node:internal/modules/esm/module_job:66:26)
    at ESMLoader.#createModuleJob (node:internal/modules/esm/loader:297:17)
    at ESMLoader.getModuleJob (node:internal/modules/esm/loader:261:34)
    at async ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:81:21)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

Version:
```
commit 5cd1b0b856501f9d7fdad8414903f0b6cdcdc001 (HEAD -> master, tag: v0.4.1, origin/master, origin/dev, origin/HEAD, dev)
Author: Jack Chakany <jack@chaker.net>
Date:   Fri May 5 08:05:36 2023 -0400

    Bump version to v0.4.1
```

**Comment by chakany** [comment]
Can I see a list of all of the files and folders inside your nblog directory? There may be missing folders for whatever reason.

**Comment by damascene** [comment]
Thank you for responding 

```
LICENSE
node_modules
package.json
playwright.config.ts
postcss.config.cjs
README.md
src
svelte.config.js
tailwind.config.cjs
tests
tsconfig.json
vite.config.ts
yarn.lock
```

**Comment by chakany** [comment]
I'm unsure if this is displaying all of the folders here, there should be many hidden folders (starting with a `.`)

Try doing `ls -a`

**Comment by damascene** [comment]
```
.     .env.example   .git        .idea         .npmrc                postcss.config.cjs  README.md         tailwind.config.cjs  vite.config.ts  .yarnrc.yml
..    .eslintignore  .github     LICENSE       package.json          .prettierignore     src               tests                .yarn
.env  .eslintrc.cjs  .gitignore  node_modules  playwright.config.ts  .prettierrc         svelte.config.js  tsconfig.json        yarn.lock

```

**Comment by chakany** [comment]
It looks like your `.svelte-kit` directory is missing, try deleting your local copy and re-follow the steps in the README.

**Comment by damascene** [comment]
It doesn't mention what method to install svelte there. This is the second time deleting the dir and trying form start but I'm getting the same error too.

**Comment by damascene** [comment]
```
$ git clone https://github.com/damascene/nblog.git
Cloning into 'nblog'...
remote: Enumerating objects: 1111, done.
remote: Counting objects: 100% (286/286), done.
remote: Compressing objects: 100% (106/106), done.
remote: Total 1111 (delta 256), reused 182 (delta 179), pack-reused 825
Receiving objects: 100% (1111/1111), 239.07 KiB | 1.56 MiB/s, done.
Resolving deltas: 100% (720/720), done.

$ cd nblog/
$ git checkout master
Branch 'master' set up to track remote branch 'master' from 'origin'.
Switched to a new branch 'master'
$ cp ../.env .
$ yarn
➤ YN0070: Migrating from Yarn 1; automatically enabling the compatibility node-modules linker 👍

➤ YN0000: ┌ Resolution step
➤ YN0032: │ fsevents@npm:2.3.2: Implicit dependencies on node-gyp are discouraged
➤ YN0032: │ utf-8-validate@npm:5.0.10: Implicit dependencies on node-gyp are discouraged
➤ YN0032: │ bufferutil@npm:4.0.7: Implicit dependencies on node-gyp are discouraged
➤ YN0061: │ @npmcli/move-file@npm:2.0.1 is deprecated: This functionality has been moved to @npmcli/fs
➤ YN0000: └ Completed in 21s 953ms
➤ YN0000: ┌ Fetch step
➤ YN0013: │ yaeti@npm:0.0.6 can't be found in the cache and will be fetched from the remote registry
➤ YN0013: │ yallist@npm:4.0.0 can't be found in the cache and will be fetched from the remote registry
➤ YN0013: │ yaml@npm:2.2.1 can't be found in the cache and will be fetched from the remote registry
➤ YN0013: │ yocto-queue@npm:0.1.0 can't be found in the cache and will be fetched from the remote registry
➤ YN0013: │ yocto-queue@npm:1.0.0 can't be found in the cache and will be fetched from the remote registry
➤ YN0000: └ Completed in 1s 121ms
➤ YN0000: ┌ Link step
➤ YN0007: │ svelte-preprocess@npm:5.0.3 [1773f] must be built because it never has been before or the last one failed
➤ YN0007: │ @fortawesome/fontawesome-common-types@npm:6.4.0 must be built because it never has been before or the last one failed
➤ YN0007: │ esbuild@npm:0.17.11 must be built because it never has been before or the last one failed
➤ YN0007: │ bufferutil@npm:4.0.7 must be built because it never has been before or the last one failed
➤ YN0007: │ es5-ext@npm:0.10.62 must be built because it never has been before or the last one failed
➤ YN0007: │ utf-8-validate@npm:5.0.10 must be built because it never has been before or the last one failed
➤ YN0007: │ @fortawesome/free-brands-svg-icons@npm:6.4.0 must be built because it never has been before or the last one failed
➤ YN0007: │ @fortawesome/free-solid-svg-icons@npm:6.4.0 must be built because it never has been before or the last one failed
➤ YN0007: │ @sveltejs/kit@npm:1.15.7 [1773f] must be built because it never has been before or the last one failed
➤ YN0000: └ Completed in 8s 62ms
➤ YN0000: Done with warnings in 31s 247ms
$ yarn build
▲ [WARNING] Cannot find base config file "./.svelte-kit/tsconfig.json" [tsconfig.json]

    tsconfig.json:2:12:
      2 │   "extends": "./.svelte-kit/tsconfig.json",
        ╵              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

error during build:
TypeError [ERR_UNKNOWN_FILE_EXTENSION]: Unknown file extension ".json" for /home/user/nostr_stuff/nblog/package.json
    at new NodeError (node:internal/errors:371:5)
    at Object.getFileProtocolModuleFormat [as file:] (node:internal/modules/esm/get_format:87:11)
    at defaultGetFormat (node:internal/modules/esm/get_format:102:38)
    at defaultLoad (node:internal/modules/esm/load:21:14)
    at ESMLoader.load (node:internal/modules/esm/loader:359:26)
    at ESMLoader.moduleProvider (node:internal/modules/esm/loader:280:58)
    at new ModuleJob (node:internal/modules/esm/module_job:66:26)
    at ESMLoader.#createModuleJob (node:internal/modules/esm/loader:297:17)
    at ESMLoader.getModuleJob (node:internal/modules/esm/loader:261:34)
    at async ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:81:21)

```

**Comment by chakany** [comment]
Well you're using the wrong version of yarn here, v1 is what should be used.

As for your issue, it seems that either yarn itself or vite cannot generate the .svelte-kit directory, maybe a permissions issue? I have no idea what is causing this or how to fix.

**Comment by damascene** [comment]
Thank you for trying to help me. I've tested both yarn `1.22.19` and `3.5.1` do you want me to try exactly version `1.0.0`?

I'm using the home directory so permission issue is unlikely. Do you have Docker file to reproduce the same setup you have or something similar?

**Comment by chakany** [comment]
No, `1.22.19` is ok.

I don't use Docker on my local machine right now, I'm running MacOS.

**Comment by damascene** [comment]
I've solved this by upgrading to node 16.20 and using `npm i` and `npm run build` instead of yarn.
